pkg/xray/resource_xray_license_policy.go

```diff
@@ -0,0 +1,168 @@
+package xray
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceXrayLicensePolicyV2() *schema.Resource {
+	return &schema.Resource{
+		SchemaVersion: 1,
+		Create:        resourceXrayPolicyCreate,
+		Read:          resourceXrayPolicyRead,
+		Update:        resourceXrayPolicyUpdate,
+		Delete:        resourceXrayPolicyDelete,
+		Description: "Creates an xray policy using V2 of the underlying APIs. Please note: " +
+			"It's only compatible with Bearer token auth method (Identity and Access => Access Tokens",
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"author": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"modified": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"rules": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"priority": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+						"criteria": {
+							Type:     schema.TypeList,
+							Required: true,
+							MinItems: 1,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"banned_licenses": {
+										Type:     schema.TypeList,
+										Optional: true,
+										Elem: &schema.Schema{
+											Type: schema.TypeString,
+										},
+									},
+									"allowed_licenses": {
+										Type:     schema.TypeList,
+										Optional: true,
+										Elem: &schema.Schema{
+											Type: schema.TypeString,
+										},
+									},
+									"allow_unknown": {
+										Type:     schema.TypeBool,
+										Optional: true,
+									},
+									"multi_license_permissive": {
+										Type:     schema.TypeBool,
+										Optional: true,
+									},
+								},
+							},
+						},
+						"actions": {
```

Comment:
I am not sure if we are supposed to use plurals with TF like this. Please find out what the standard is

Reply:
"actions" here are legit because it is always a list of actions and you can only have one per TF resource.
Also, it has the same name in the API call body, and I think it's better to have the names identical to the JSON body fields names.